### PR TITLE
Add symlink metadata to FileArtifactValue

### PR DESCRIPTION
**Pull request by buchgr**
This change makes skyframe track symlinks in output files and
invalidate actions if symlink targets change. This change is
a preqrequisite to encoding symlink information in the open
source remote execution protocol.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java [hunk 1]
```diff
@@ -177,36 +179,49 @@ public static FileArtifactValue create(Artifact artifact, FileValue fileValue)
       throws IOException {
     boolean isFile = fileValue.isFile();
     FileContentsProxy proxy = getProxyFromFileStateValue(fileValue.realFileStateValue());
-    return create(
+    FileArtifactValue resolvedValue = create(
```
Comment (lberki) [inline src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java]: Will this change the behavior of symlinks-to-directories? If not, where is `isFile` set to false in that case?

**Comment by buchgr** [comment]
@lberki @ericfelly I just now found out about an issue with this change. Apparently, FileArtifactValue always needs to represent the resolved file/directory and must never return true for `isSymlink`. So please hold off from reviewing this for now. Sorry about that.

**Comment by lberki** [comment]
Well, I already did, but how did you find it *now*?

**Comment by buchgr** [comment]
@lberki thanks a lot for your review!! I talked to @ulfjack offline. He told me that `FileArtifactValue` always needs to represent the resolved value. That's apparently a major difference between `FileStateValue` and `FileArtifactValue`. Thus, I believe this change is not going to fly.

**Pull request closed without merging**